cmd/saml2aws/commands/console.go

```diff
@@ -0,0 +1,128 @@
+package commands
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/skratchdot/open-golang/open"
+	"github.com/versent/saml2aws/pkg/awsconfig"
+	"github.com/versent/saml2aws/pkg/flags"
+)
+
+// Exec execute the supplied command after seeding the environment
+func Console(consoleFlags *flags.LoginExecFlags) error {
+
+	account, err := buildIdpAccount(consoleFlags)
+	if err != nil {
+		return errors.Wrap(err, "error building login details")
+	}
+
+	sharedCreds := awsconfig.NewSharedCredentials(account.Profile)
+
+	// this checks if the credentials file has been created yet
+	// can only really be triggered if saml2aws exec is run on a new
+	// system prior to creating $HOME/.aws
+	exist, err := sharedCreds.CredsExists()
+	if err != nil {
+		return errors.Wrap(err, "error loading credentials")
+	}
+	if !exist {
+		fmt.Println("unable to load credentials, login required to create them")
+		return nil
+	}
+
+	awsCreds, err := sharedCreds.Load()
+	if err != nil {
+		return errors.Wrap(err, "error loading credentials")
+	}
+
+	if awsCreds.Expires.Sub(time.Now()) < 0 {
+		return errors.New("error aws credentials have expired")
+	}
+
+	ok, err := checkToken(account.Profile)
+	if err != nil {
+		return errors.Wrap(err, "error validating token")
+	}
+
+	if !ok {
+		err = Login(consoleFlags)
+	}
+	if err != nil {
+		return errors.Wrap(err, "error logging in")
+	}
+
+	if consoleFlags.ExecProfile != "" {
```

Comment:
```
saml2aws console --exec-profile staging-admin
saml2aws: error: unknown long flag '--exec-profile', try --help
```

Does this command need a different flag set? It looks like it is currently sharing with 'login'

Reply:
Could be a copy paste issue, i wanted to demo it more widely so I released it a bit early TBH  given it was a new command.

It may have been a bit rough.

Reply:
I'm not 100% sure you can do a 'login to this secondary assume' (eg: okta -> loginRole -> assumedRole -> web console) anyway - I tried to do it using python and got stuck - the closest I got is that you can open directly to the 'Switch Role' page directly and fill the info in, which is at least just a click away.